src/simstation/Agent.java

```diff
@@ -1,13 +1,94 @@
 package simstation;
 
-public class Agent {
+// Assuming Heading is an enum already defined somewhere
+import mvc.Utilities;
 
+import java.io.Serializable;
+
+public abstract class Agent implements Runnable, Serializable {
+
+    private String name;
     protected Heading heading;
+    private int xc, yc; // Presumably for coordinates, but not shown in your diagram
+    private boolean suspended = false;
+    private boolean stopped = false;
+    private transient Thread myThread; // Transient because Threads are not serializable
+
     public Agent() {
+        this.heading = Heading.random(); // Assign a random heading
+        this.xc = Utilities.rng.nextInt(); // Example usage, needs proper range
+        this.yc = Utilities.rng.nextInt(); // Example usage, needs proper range
+    }
+
+
+
+    public String getName() {
+        return name;
+    }
+
+    public synchronized void start() {
+        stopped = false;
+        suspended = false;
+        myThread = new Thread(this);
+        myThread.start();
+    }
+
+    public synchronized void suspend() {
+        suspended = true;
+    }
+
+    public synchronized void resume() {
+        if (suspended) {
+            suspended = false;
+            notify();
+        }
+    }
+
+    public synchronized void stop() {
+        stopped = true;
+        if (myThread != null) {
```

Comment:
this is normally taken care of in the run method. Here you only need to change the stopped value